### PR TITLE
Python: use google.api.core.Operation for long running operations.

### PR DESCRIPTION
**Pull request by landrito**
Updates #1320 

Here is the Operation class for reference:
https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/core/google/api/core/operation.py

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/py/python_setup_library.baseline [hunk 1]
```diff
@@ -11,6 +11,7 @@ import sys
 
 install_requires = [
     'google-auth>=1.0.2, <2.0dev',
+    'google-cloud-core>=0.27.0, <1.0dev',
```
Comment (lukesneeringer) [inline src/test/java/com/google/api/codegen/testdata/py/python_setup_library.baseline]: `< 0.28dev`

Comment (landrito) [reply src/test/java/com/google/api/codegen/testdata/py/python_setup_library.baseline]: Done.

File: src/main/java/com/google/api/codegen/viewmodel/DynamicLangXApiView.java [hunk 2]
```diff
@@ -131,6 +131,9 @@ public boolean hasMissingDefaultOptions() {
   @Nullable
   public abstract String gapicPackageName();
 
+  @Nullable
+  public abstract List<String> operationTypes();
```
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/viewmodel/DynamicLangXApiView.java]: What's this for?

Comment (landrito) [reply src/main/java/com/google/api/codegen/viewmodel/DynamicLangXApiView.java]: Woops, Theres two implementations of operation in google-cloud-core. I accidentally implemented the wrong one and this is a byproduct. Removing.

Comment (landrito) [reply src/main/java/com/google/api/codegen/viewmodel/DynamicLangXApiView.java]: Fixed.

**Comment by landrito** [comment]
This is now obsolete.

**Pull request closed without merging**